pkg/vfs/download_store.go

```diff
@@ -0,0 +1,92 @@
+package vfs
+
+import (
+	"encoding/hex"
+	"time"
+
+	"github.com/cozy/cozy-stack/pkg/crypto"
+)
+
+// A DownloadStore is essentially an object to store Archives & Files by keys
+type DownloadStore interface {
+	AddFile(f string) (string, error)
+	AddArchive(a *Archive) (string, error)
+	GetFile(k string) (string, error)
+	GetArchive(k string) (*Archive, error)
+}
+
+type fileRef struct {
+	Path      string
+	ExpiresAt time.Time
+}
+
+// downloadStoreTTL is the time an Archive stay alive
+const downloadStoreTTL = 1 * time.Hour
+
+var storeStore map[string]*memStore
+
+// GetStore returns the DownloadStore for the given Instance
+func GetStore(domain string) DownloadStore {
+	if storeStore == nil {
+		storeStore = make(map[string]*memStore)
+	}
+	store, ok := storeStore[domain]
+	if ok {
+		return store
+	}
+	storeStore[domain] = &memStore{
+		Archives: make(map[string]*Archive),
+		Files:    make(map[string]*fileRef),
+	}
+	return storeStore[domain]
+}
+
+type memStore struct {
+	Archives map[string]*Archive
+	Files    map[string]*fileRef
+}
+
+func (s *memStore) makeSecret() string {
+	return hex.EncodeToString(crypto.GenerateRandomBytes(16))
+}
+
+func (s *memStore) AddFile(f string) (string, error) {
+	fref := &fileRef{
```

Comment:
Maybe before adding a file, we should remove all the filesRef that have expired in `s.Files` to avoid memory leak